File: object_detection/src/static_transform_publisher.cpp
#include <rclcpp/rclcpp.hpp>
#include <tf2_ros/static_transform_broadcaster.h>
#include <geometry_msgs/msg/transform_stamped.hpp>
#include <tf2/LinearMath/Quaternion.h>

class StaticTransformPublisher : public rclcpp::Node
{
public:
    StaticTransformPublisher() : Node("static_transform_publisher")
    {
        tf_static_broadcaster_ = std::make_shared<tf2_ros::StaticTransformBroadcaster>(this);
        publishStaticTransform();
        RCLCPP_INFO(this->get_logger(), "Static transform publisher initialized (C++)");
    }

private:
    void publishStaticTransform()
    {
        geometry_msgs::msg::TransformStamped static_transform;
        static_transform.header.stamp = this->get_clock()->now();
        static_transform.header.frame_id = "base_link";
        static_transform.child_frame_id = "wrist_rgbd_camera_depth_optical_frame";
        
        static_transform.transform.translation.x = 0.3;
        static_transform.transform.translation.y = 0.0;
        static_transform.transform.translation.z = 0.4;
        
        tf2::Quaternion quat;
        quat.setRPY(1.5708, 0, 0);
        static_transform.transform.rotation.x = quat.x();
        static_transform.transform.rotation.y = quat.y();
        static_transform.transform.rotation.z = quat.z();
        static_transform.transform.rotation.w = quat.w();
        
        tf_static_broadcaster_->sendTransform(static_transform);
        RCLCPP_INFO(this->get_logger(), "Published static transform");
    }
    
    std::shared_ptr<tf2_ros::StaticTransformBroadcaster> tf_static_broadcaster_;
};

int main(int argc, char** argv)
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<StaticTransformPublisher>());
    rclcpp::shutdown();
    return 0;
}

File: object_detection/scripts/static_transform_publisher.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from tf2_ros import StaticTransformBroadcaster
from geometry_msgs.msg import TransformStamped
import tf_transformations
import math

class StaticTransformPublisher(Node):
    def __init__(self):
        super().__init__('static_transform_publisher')
        
        self.tf_static_broadcaster = StaticTransformBroadcaster(self)
        self.publish_static_transform()
        self.get_logger().info('Static transform publisher initialized (Python)')

    def publish_static_transform(self):
        static_transform = TransformStamped()
        static_transform.header.stamp = self.get_clock().now().to_msg()
        static_transform.header.frame_id = 'base_link'
        static_transform.child_frame_id = 'wrist_rgbd_camera_depth_optical_frame'
        
        # Use the ACTUAL transform values from tf2_echo
        static_transform.transform.translation.x = 0.338
        static_transform.transform.translation.y = 0.450
        static_transform.transform.translation.z = 0.100
        
        # Use the ACTUAL rotation quaternion from tf2_echo
        static_transform.transform.rotation.x = 0.000
        static_transform.transform.rotation.y = 0.866
        static_transform.transform.rotation.z = -0.500
        static_transform.transform.rotation.w = 0.000
        
        self.tf_static_broadcaster.sendTransform(static_transform)
        self.get_logger().info('Published static transform with ACTUAL camera position and orientation')
        self.get_logger().info('Translation: [0.338, 0.450, 0.100]')
        self.get_logger().info('RPY: [-120°, 0°, 180°]')

def main(args=None):
    rclpy.init(args=args)
    node = StaticTransformPublisher()
    
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()

if __name__ == '__main__':
    main()